source/postprocess/visualization/heating.cc

```diff
@@ -120,17 +125,31 @@ namespace aspect
 
             for (unsigned int c=0; c<this->n_compositional_fields(); ++c)
               in.composition[q][c] = uh[q][this->introspection().component_indices.compositional_fields[c]];
+
+            mid_point += evaluation_points[q]/n_quadrature_points;
           }
 
         in.position = evaluation_points;
 
-        this->get_material_model().evaluate(in, out);
+        typename DoFHandler<dim>::active_cell_iterator cell;
+        cell = (GridTools::find_active_cell_around_point<> (this->get_mapping(), this->get_dof_handler(), mid_point)).first;
```

Comment:
The problem is here that we need the melt velocity. So we need the cell to get it from the solution vector (it is not in the material model inputs, because the material model does not need it). If we do not pass the cell, it takes the melt velocity as zero, and that means that the visualization output is wrong.